cmd/id-exporter/main_test.go

```diff
@@ -61,24 +61,73 @@ func TestFindIDs(t *testing.T) {
 	// *not* be present since their certificate has already expired. Unlike
 	// previous versions of this test RegD is not filtered out for having a `tel:`
 	// contact field anymore - this is the duty of the notify-mailer.
-	ids, err = testCtx.c.findIDs()
+	results, err = testCtx.c.findIDs()
 	test.AssertNotError(t, err, "findIDs() produced error")
-	test.AssertEquals(t, len(ids), 3)
-	test.AssertEquals(t, ids[0].ID, regA.ID)
-	test.AssertEquals(t, ids[1].ID, regC.ID)
-	test.AssertEquals(t, ids[2].ID, regD.ID)
+	test.AssertEquals(t, len(results), 3)
+	test.AssertEquals(t, results[0].ID, regA.ID)
+	test.AssertEquals(t, results[1].ID, regC.ID)
+	test.AssertEquals(t, results[2].ID, regD.ID)
 
 	// Allow a 1 year grace period
 	testCtx.c.grace = 360 * 24 * time.Hour
-	ids, err = testCtx.c.findIDs()
+	results, err = testCtx.c.findIDs()
 	test.AssertNotError(t, err, "findIDs() produced error")
 	// Now all four registration should be returned, including RegB since its
 	// certificate expired within the grace period
-	test.AssertEquals(t, len(ids), 4)
-	test.AssertEquals(t, ids[0].ID, regA.ID)
-	test.AssertEquals(t, ids[1].ID, regB.ID)
-	test.AssertEquals(t, ids[2].ID, regC.ID)
-	test.AssertEquals(t, ids[3].ID, regD.ID)
+	test.AssertEquals(t, len(results), 4)
+	test.AssertEquals(t, results[0].ID, regA.ID)
+	test.AssertEquals(t, results[1].ID, regB.ID)
+	test.AssertEquals(t, results[2].ID, regC.ID)
+	test.AssertEquals(t, results[3].ID, regD.ID)
+}
+
+func TestFindIDsWithExampleHostnames(t *testing.T) {
+	testCtx := setup(t)
+	defer testCtx.cleanUp()
+
+	// Add some test registrations
+	testCtx.addRegistrations(t)
+
+	// Run findIDsWithExampleHostnames - since no certificates have been
+	// added corresponding to the above registrations, no IDs should be
+	// found.
+	results, err := testCtx.c.findIDsWithExampleHostnames()
+	test.AssertNotError(t, err, "findIDs() produced error")
+	test.AssertEquals(t, len(results), 0)
+
+	// Now add some certificates
+	testCtx.addCertificates(t)
+
+	// Run findIDsWithExampleHostnames - since there are three
+	// registrations with unexpired certs we should get exactly three
+	// IDs back: RegA, RegC and RegD. RegB should *not* be present since
+	// their certificate has already expired.
+	results, err = testCtx.c.findIDsWithExampleHostnames()
+	test.AssertNotError(t, err, "findIDs() produced error")
+	test.AssertEquals(t, len(results), 3)
+	test.AssertEquals(t, results[0].ID, regA.ID)
```

Comment:
nit: It'll be fine of course, but it always makes me nervous when we depend on order in tests but are relying on defaults to produce that ordering. Since this is depending on `GROUP BY ... ASC`, we could be explicit. See [GROUP BY](https://mariadb.com/kb/en/group-by/).

Same is true for the other queries, of course, but it's kinda worse, because whereas via GROUP BY it's actually already happening and we're worried about the default changing, for the others where no order is specified, it's actually not specified how MariaDB will respond, and adding an ORDER BY to those queries is almost certainly going to slow them down, so maybe it's best to just ignore this nit and call it good-as-is.

Reply:
Really good point, I'll just rewrite these tests to use a case switch statement instead.